### PR TITLE
Preserve the SIMPLE_CI environment variable when executing the tests as user 'ohpc'

### PR DESCRIPTION
**Pull request by martin-g**
Add mapping for FFTW test.
Fix the value of TASKS depending on whether or not SIMPLE_CI is exported.

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

### CODE REVIEW

**Comment by martin-g** [comment]
RHEL and Rocky (8.6) do not support `-w`:
```
su --help

Usage:
 su [options] [-] [<user> [<argument>...]]

Change the effective user ID and group ID to that of <user>.
A mere - implies -l.  If <user> is not given, root is assumed.

Options:
 -m, -p, --preserve-environment  do not reset environment variables
 -g, --group <group>             specify the primary group
 -G, --supp-group <group>        specify a supplemental group

 -, -l, --login                  make the shell a login shell
 -c, --command <command>         pass a single command to the shell with -c
 --session-command <command>     pass a single command to the shell with -c
                                   and do not create a new session
 -f, --fast                      pass -f to the shell (for csh or tcsh)
 -s, --shell <shell>             run <shell> if /etc/shells allows it
 -P, --pty                       create a new pseudo-terminal

 -h, --help                      display this help
 -V, --version                   display version
```

I will rework it to use `sudo` instead.

File: tests/ci/spec_to_test_mapping.py [hunk 1]
```diff
@@ -42,6 +42,10 @@
         'scotch',
         'zlib-devel'
     ],
+    '../components/parallel-libs/fftw/SPECS/fftw.spec': [
+        'fftw',
+        'openmpi4-gnu12-ohpc mpich-gnu12-ohpc'
+    ],
```
Comment (martin-g) [inline tests/ci/spec_to_test_mapping.py:48]: Should I add entries also for all tests whose `rm_execution` has been modified below ?

Comment (martin-g) [reply tests/ci/spec_to_test_mapping.py:48]: I'll do it!

File: tests/ci/spec_to_test_mapping.py [hunk 2]
```diff
@@ -42,6 +42,10 @@
         'scotch',
         'zlib-devel'
     ],
+    '../components/parallel-libs/fftw/SPECS/fftw.spec': [
```
Comment (martin-g) [inline tests/ci/spec_to_test_mapping.py]: The leading `../` will be removed in the next commit!
It seems this typo didn't break the test execution ...

File: tests/mpi/tests/rm_execution_multi_host [hunk 3]
```diff
@@ -12,7 +12,7 @@ check_rms
 RMS=$RESOURCE_MANAGER
 
 NODES=2
-TASKS=8
+TASKS=`tasks_count 8`
```
Comment (martin-g) [inline tests/mpi/tests/rm_execution_multi_host:15]: Which .spec should be added to `spec_to_test_mapping.py` for this test ?

File: tests/mpi/tests/rm_execution_single_host [hunk 4]
```diff
@@ -12,7 +12,7 @@ check_rms
 RMS=$RESOURCE_MANAGER
 
 NODES=1
-TASKS=2
+TASKS=`tasks_count 2`
```
Comment (martin-g) [inline tests/mpi/tests/rm_execution_single_host:15]: Which .spec should be added to `spec_to_test_mapping.py` for this test ?

File: tests/rms-harness/tests/test_harness [hunk 5]
```diff
@@ -12,7 +12,7 @@ check_rms
 rm=$RESOURCE_MANAGER
 
 NODES=2
-TASKS=2
+TASKS=`tasks_count 2`
```
Comment (martin-g) [inline tests/rms-harness/tests/test_harness:15]: Which .spec should be added to spec_to_test_mapping.py for this test ?

File: tests/ci/spec_to_test_mapping.py (hunk 2 above)
Comment (martin-g) [reply tests/ci/spec_to_test_mapping.py]: Answering to myself: If the key is misspelled then `setup_slurm_and_run_tests.sh` cannot find it and it does not execute any tests.

File: tests/rms-harness/tests/test_harness (hunk 5 above)
Comment (adrianreber) [reply tests/rms-harness/tests/test_harness:15]: maybe always?

File: tests/mpi/tests/rm_execution_single_host (hunk 4 above)
Comment (adrianreber) [reply tests/mpi/tests/rm_execution_single_host:15]: probably openmpi4.spec, mpich.spec and mvapich2.spec

File: tests/mpi/tests/rm_execution_multi_host (hunk 3 above)
Comment (adrianreber) [reply tests/mpi/tests/rm_execution_multi_host:15]: see the other mpi test

File: tests/mpi/tests/rm_execution_single_host (hunk 4 above)
Comment (martin-g) [reply tests/mpi/tests/rm_execution_single_host:15]: I've added mappings for `openmpi4` and `mpich`.
`mvapich2` requires InfiniBand and thus cannot be tested on Github Actions.

File: tests/rms-harness/tests/test_harness (hunk 5 above)
Comment (martin-g) [reply tests/rms-harness/tests/test_harness:15]: I.e. to add `--enable-rms-harness` at https://github.com/martin-g/openeuler-ohpc/blob/preserve_SIMPLE_CI_for_test_execution/tests/ci/setup_slurm_and_run_tests.sh#L92 ?

Comment (adrianreber) [reply tests/rms-harness/tests/test_harness:15]: Yes, something like this. Not sure myself.

Comment (martin-g) [reply tests/rms-harness/tests/test_harness:15]: Done!
We can remove it if it causes any problems!

Comment (martin-g) [reply tests/rms-harness/tests/test_harness:15]: It failed on RHEL.
It passes fine here (on openEuler).
I'm trying to find the reason.

Comment (martin-g) [reply tests/rms-harness/tests/test_harness:15]: The error is:
```
-------------------------------------------------------
User Environment: RMS Harness tests: gnu12-openmpi4
-------------------------------------------------------
Lmod has detected the following error: The following module(s) are unknown:
"prun"
```

File: tests/ci/setup_slurm_and_run_tests.sh [hunk 6]
```diff
@@ -12,14 +12,15 @@ dnf -y install \
...
+	sudo \
 	slurm-slurmd-ohpc \
 	slurm-slurmctld-ohpc \
 	slurm-example-configs-ohpc \
 	slurm-ohpc
 
 # Install rebuilt packages (if any)
 # shellcheck disable=SC2046 # (we want the words to be split)
-dnf -y install $(find /home/"${USER}"/rpmbuild/RPMS/ -name "*rpm") || true
+dnf -y install prun-ohpc gnu12-compilers-ohpc openmpi4-gnu12-ohpc mpich-gnu12-ohpc $(find /home/"${USER}"/rpmbuild/RPMS/ -name "*rpm") || true
```
Comment (martin-g) [inline tests/ci/setup_slurm_and_run_tests.sh:23]: Those are needed by RMS Harness.
Is it OK to pre-install them here ?

**Comment by adrianreber** [comment]
That is a big PR and I hope it does not break anything. It looks good from the CI point of view. I will merge it. Thanks.

**Pull request merged**